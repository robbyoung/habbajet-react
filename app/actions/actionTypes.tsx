export enum ActionType {
    ADD_HABBAJET = 'Add Habbajet',
    ADD_HABIT_RESULT = 'Add Habit Result',
    UPDATE_BUDGET = 'Update Budget',
    RESET_HABBAJET = 'Reset Habbajet',
    LOAD_STATE = 'Load State',
    SELECT_HABBAJET = 'Select Habbajet',
    UPDATE_EDITOR_FIELD = 'Update Editor Field',
    VALIDATE_EDITOR = 'Validate Editor',
    CLEAR_EDITOR = 'Clear Editor',
    ADD_PURCHASE = 'Add Purchase',
    UPDATE_PURCHASE_EDITOR = 'Update Purchase Editor',
    CLEAR_PURCHASE_EDITOR = 'Clear Purchase Editor',
    VALIDATE_PURCHASE_EDITOR = 'Validate Purchase Editor',
    SET_HABBAJET_TO_EDIT = 'Set Habbajet To Edit',
    DELETE_HABBAJET = 'Delete Habbajet',
    ADD_TAG = 'Add Tag',
    SET_PURCHASE_TO_EDIT = 'Set Purchase To Edit',
    SET_TAG_TO_EDIT = 'Set Tag To Edit',
    UPDATE_TAG_EDITOR = 'Update Tag Editor',
    CLEAR_TAG_EDITOR = 'Clear Tag Editor',
    VALIDATE_TAG_EDITOR = 'Validate Tag Editor',
    DELETE_TAG = 'Delete Tag',
}
