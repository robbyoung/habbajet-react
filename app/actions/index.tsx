export * from './actionTypes';
export * from './addHabbajet';
export * from './addHabitResult';
export * from './updateBudget';
export * from './resetHabbajet';
export * from './loadState';
export * from './selectHabbajet';
export * from './updateEditorField';
export * from './validateEditor';
export * from './clearEditor';
