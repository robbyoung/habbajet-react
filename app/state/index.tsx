export enum HabitResult {
    Success,
    SlackSuccess,
    Failure,
}

export interface Habbajet {
    id: string;
    name: string;
    maxValue: number;
    currentValue: number;
    modifier: number;
    results: HabitResult[];
    color: string;
    date: string;
    toClaim: boolean;
    selected: boolean;
    currentStreak: number;
    bestStreak: number;
    oldStreaks: [number, number];
    totalSlack: number;
    remainingSlack: number;
}

export interface Purchase {
    id: string;
    name: string;
    cost: number;
    date: string;
    tagId: string;
}

export interface EditorField {
    value: string;
    errorMessage: string;
}

export interface HabbajetEditor {
    name: EditorField;
    value: EditorField;
    modifier: EditorField;
    slack: EditorField;
    color: string;
    validated: boolean;
}

export interface PurchaseEditor {
    name: EditorField;
    cost: EditorField;
    tagId: string;
    validated: boolean;
    id?: string;
}

export interface Tag {
    id: string;
    name: string;
    color: string;
}

export interface State {
    habbajets: Habbajet[];
    purchases: Purchase[];
    tags: Tag[];
    habbajetEditor: HabbajetEditor;
    purchaseEditor: PurchaseEditor;
    budget: number;
}
