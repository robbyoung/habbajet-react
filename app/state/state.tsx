export interface Habbajet {
    name: string;
    value: number;
    days: boolean[];
}

export interface Purchase {
    name: string;
    value: number;
}

export interface State {
    habits: Habbajet;
    purchages: Purchase;
    budget: string;
}
