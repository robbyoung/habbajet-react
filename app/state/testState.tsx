import {
    State,
    Habbajet,
    Purchase,
    HabbajetEditor,
    Tag,
    PurchaseEditor,
    TagEditor,
} from '.';
import {habbajetColors} from '../colors';

export function createTestState(
    numHabbajets: number,
    numPurchases: number,
    budget: number,
    selectedIndex: number = -1,
): State {
    const habbajets: Habbajet[] = [];
    for (let i = 0; i < numHabbajets; i++) {
        habbajets.push({
            id: `${i}`,
            name: `Habbajet ${i}`,
            maxValue: 100,
            currentValue: 100,
            modifier: 2,
            results: [],
            color: habbajetColors[i % habbajetColors.length],
            date: '2020-03-22T11:00:00.000Z',
            toClaim: false,
            selected: i === selectedIndex,
            currentStreak: 0,
            bestStreak: 5,
            oldStreaks: [0, 5],
            totalSlack: 0,
            remainingSlack: 0,
            description: '',
        });
    }

    const purchases: Purchase[] = [];
    for (let i = 0; i < numPurchases; i++) {
        purchases.push({
            id: `${i}`,
            name: `Purchase ${i}`,
            cost: i,
            date: '2020-03-22T11:00:00.000Z',
            tagId: `${i % 10}`,
        });
    }

    const tags: Tag[] = [];
    for (let i = 0; i < 10; i++) {
        tags.push({
            id: `${i}`,
            name: `Tag ${i}`,
            color: habbajetColors[i % habbajetColors.length],
        });
    }

    const EMPTY_FIELD = {
        value: '',
        errorMessage: '',
    };
    const habbajetEditor: HabbajetEditor = {
        name: {...EMPTY_FIELD},
        description: {...EMPTY_FIELD},
        value: {...EMPTY_FIELD},
        modifier: {...EMPTY_FIELD},
        slack: {...EMPTY_FIELD},
        color: habbajetColors[0],
        validated: false,
    };

    const purchaseEditor: PurchaseEditor = {
        name: {...EMPTY_FIELD},
        cost: {...EMPTY_FIELD},
        tagId: '',
        validated: false,
    };

    const tagEditor: TagEditor = {
        name: {...EMPTY_FIELD},
        color: habbajetColors[0],
        validated: false,
    };

    return {
        habbajets,
        purchases,
        budget,
        habbajetEditor,
        purchaseEditor,
        tagEditor,
        tags,
    };
}

export function createTestEditor(
    name: string,
    value: string,
    modifier: string,
    slack = '0',
    validated = false,
): HabbajetEditor {
    return {
        name: {
            value: name,
            errorMessage: '',
        },
        description: {
            value: '',
            errorMessage: '',
        },
        value: {
            value,
            errorMessage: '',
        },
        modifier: {
            value: modifier,
            errorMessage: '',
        },
        slack: {
            value: slack,
            errorMessage: '',
        },
        color: habbajetColors[0],
        validated,
    };
}

export function createTestPurchaseEditor(
    name: string,
    value: string,
    validated = false,
): PurchaseEditor {
    return {
        name: {
            value: name,
            errorMessage: '',
        },
        cost: {
            value,
            errorMessage: '',
        },
        tagId: '0',
        validated,
    };
}

export function createTestTagEditor(
    name: string,
    validated = false,
): TagEditor {
    return {
        name: {
            value: name,
            errorMessage: '',
        },
        color: '#000000',
        validated,
    };
}
