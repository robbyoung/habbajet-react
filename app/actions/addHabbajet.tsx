import moment from 'moment';
import {Action} from 'redux';
import {ActionType} from './actionTypes';
import {Habbajet} from '../state';

export interface AddHabbajetAction extends Action {
    type: ActionType.ADD_HABBAJET;
    newHabbajet: Habbajet;
}

export function addHabbajetAction(
    name: string,
    value: number,
    modifier: number,
    slack: number,
    color: string,
): AddHabbajetAction {
    const monday = moment()
        .startOf('isoWeek')
        .toISOString();
    return {
        type: ActionType.ADD_HABBAJET,
        newHabbajet: {
            name,
            maxValue: value,
            currentValue: value / Math.pow(modifier, 7),
            modifier,
            results: [],
            color,
            date: monday,
            toClaim: false,
            selected: false,
            currentStreak: 0,
            bestStreak: 0,
            totalSlack: slack,
            remainingSlack: slack,
        },
    };
}

export function addHabbajet(
    state: Habbajet[],
    action: AddHabbajetAction,
): Habbajet[] {
    const replaceIndex = state.findIndex(
        habbajet => habbajet.name === action.newHabbajet.name,
    );

    if (replaceIndex === -1) {
        return [...state, action.newHabbajet];
    }

    const newState = [...state];
    const edited = action.newHabbajet;
    const original = state[replaceIndex];

    newState[replaceIndex] = {
        ...original,
        name: edited.name,
        maxValue: edited.maxValue,
        modifier: edited.modifier,
        totalSlack: edited.totalSlack,
        color: edited.color,
        selected: true,
    };

    return newState;
}
