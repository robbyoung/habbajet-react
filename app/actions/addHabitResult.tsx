import moment from 'moment';
import {Action} from 'redux';
import {ActionType} from './actionTypes';
import {Habbajet, HabitResult} from '../state';

export interface AddHabitResultAction extends Action {
    type: ActionType.ADD_HABIT_RESULT;
    habbajetName: string;
    result: boolean;
}

export function addHabitResultAction(
    habbajetName: string,
    result: boolean,
): AddHabitResultAction {
    return {
        type: ActionType.ADD_HABIT_RESULT,
        habbajetName,
        result,
    };
}

export function addHabitResult(
    state: Habbajet[],
    action: AddHabitResultAction,
): Habbajet[] {
    const index = state.findIndex(
        habbajet => habbajet.name === action.habbajetName,
    );
    if (index === -1) {
        return state;
    }

    const newState = [...state];
    const edited: Habbajet = {...newState[index]};
    const date = moment(edited.date);

    date.add(1, 'day');
    if (date.day() === 1) {
        edited.toClaim = true;
    }
    edited.date = date.toISOString();

    let newResult = action.result ? HabitResult.Success : HabitResult.Failure;
    if (edited.remainingSlack > 0 && !action.result) {
        newResult = HabitResult.SlackSuccess;
    }
    edited.results = [...edited.results, newResult];

    const isSuccess = action.result || edited.remainingSlack > 0;
    if (isSuccess) {
        edited.currentStreak++;
        if (edited.currentStreak > edited.bestStreak) {
            edited.bestStreak = edited.currentStreak;
        }
        if (!action.result) {
            edited.remainingSlack--;
        }
    } else {
        edited.currentValue /= edited.modifier;
        edited.currentStreak = 0;
    }

    newState[index] = edited;
    return newState;
}
