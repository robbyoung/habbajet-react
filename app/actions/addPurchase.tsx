import moment from 'moment';
import {Action} from 'redux';
import {ActionType} from './actionTypes';
import {Purchase} from '../state';

export interface AddPurchaseAction extends Action {
    type: ActionType.ADD_PURCHASE;
    newPurchase: Purchase;
}

export function addPurchaseAction(
    name: string,
    cost: number,
    tagId: string,
): AddPurchaseAction {
    return {
        type: ActionType.ADD_PURCHASE,
        newPurchase: {
            name,
            cost,
            tagId,
            date: moment().toISOString(),
        },
    };
}

export function addPurchase(
    state: Purchase[],
    action: AddPurchaseAction,
): Purchase[] {
    return [action.newPurchase, ...state];
}
