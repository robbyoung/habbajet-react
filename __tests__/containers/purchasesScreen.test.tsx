import React from 'react';
import renderer, {act} from 'react-test-renderer';
import store from '../../app/store';
import {Provider} from 'react-redux';
import PurchasesScreen from '../../app/containers/purchasesScreen';
import {createTestState} from '../../app/state/testState';
import {loadStateAction} from '../../app/actions';
import {render, fireEvent, wait} from '@testing-library/react-native';

const mockGoToEditPurchase = jest.fn();
jest.mock('../../app/navigation/navigation', () => ({
    goToEditPurchase: () => mockGoToEditPurchase(),
}));

jest.mock('@fortawesome/react-native-fontawesome', () => ({
    FontAwesomeIcon: '',
}));

jest.mock('../../app/storage', () => ({}));

describe('New Purchases Screen Component', () => {
    it('will map state to a list of purchases', () => {
        const state = createTestState(5, 10, 200);
        store.dispatch(loadStateAction(state));

        const component = renderer.create(
            <Provider store={store}>
                <PurchasesScreen />
            </Provider>,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('display a message if there are no purchases', () => {
        act(() => {
            const state = createTestState(5, 0, 200);
            store.dispatch(loadStateAction(state));
        });

        const component = renderer.create(
            <Provider store={store}>
                <PurchasesScreen />
            </Provider>,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('touching a purchase will direct you to the edit purchase screen', async () => {
        act(() => {
            const state = createTestState(0, 10, 0);
            store.dispatch(loadStateAction(state));
        });

        const {getByTestId} = render(
            <Provider store={store}>
                <PurchasesScreen />
            </Provider>,
        );

        const row = getByTestId('purchase-0');
        fireEvent.press(row);
        await wait(() => expect(mockGoToEditPurchase).toBeCalled());
    });
});
