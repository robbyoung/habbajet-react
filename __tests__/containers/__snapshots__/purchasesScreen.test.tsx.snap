// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Purchases Screen Component can be empty if there are no purchases 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": "100%",
    }
  }
>
  <View />
</RCTScrollView>
`;

exports[`New Purchases Screen Component will map state to a list of purchases 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": "100%",
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#f5f5f5",
          "borderWidth": 2,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          Purchase 0
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          $0.00
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
              "padding": 3,
              "textAlign": "right",
            }
          }
        >
          23/03/2020
        </Text>
        <View />
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#f5f5f5",
          "borderWidth": 2,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          Purchase 1
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          $1.00
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
              "padding": 3,
              "textAlign": "right",
            }
          }
        >
          23/03/2020
        </Text>
        <View />
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#f5f5f5",
          "borderWidth": 2,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          Purchase 2
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          $2.00
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
              "padding": 3,
              "textAlign": "right",
            }
          }
        >
          23/03/2020
        </Text>
        <View />
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#f5f5f5",
          "borderWidth": 2,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          Purchase 3
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          $3.00
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
              "padding": 3,
              "textAlign": "right",
            }
          }
        >
          23/03/2020
        </Text>
        <View />
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#f5f5f5",
          "borderWidth": 2,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          Purchase 4
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          $4.00
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
              "padding": 3,
              "textAlign": "right",
            }
          }
        >
          23/03/2020
        </Text>
        <View />
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#f5f5f5",
          "borderWidth": 2,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          Purchase 5
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          $5.00
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
              "padding": 3,
              "textAlign": "right",
            }
          }
        >
          23/03/2020
        </Text>
        <View />
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#f5f5f5",
          "borderWidth": 2,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          Purchase 6
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          $6.00
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
              "padding": 3,
              "textAlign": "right",
            }
          }
        >
          23/03/2020
        </Text>
        <View />
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#f5f5f5",
          "borderWidth": 2,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          Purchase 7
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          $7.00
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
              "padding": 3,
              "textAlign": "right",
            }
          }
        >
          23/03/2020
        </Text>
        <View />
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#f5f5f5",
          "borderWidth": 2,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          Purchase 8
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          $8.00
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
              "padding": 3,
              "textAlign": "right",
            }
          }
        >
          23/03/2020
        </Text>
        <View />
      </View>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#f5f5f5",
          "borderWidth": 2,
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          Purchase 9
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "Abel",
              "fontSize": 30,
            }
          }
        >
          $9.00
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#959595",
              "fontFamily": "Abel",
              "fontSize": 20,
              "padding": 3,
              "textAlign": "right",
            }
          }
        >
          23/03/2020
        </Text>
        <View />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
